src/time.rs

```diff
@@ -335,15 +339,12 @@ impl TryConvert for chrono::DateTime<chrono::Utc> {
             timespec = rb_time_timespec(val.as_rb_value());
             Ruby::get_unchecked().qnil()
         })?;
-        if timespec.tv_sec >= 0 && timespec.tv_nsec >= 0 {
-            let mut duration = Duration::from_secs(timespec.tv_sec as _);
-            duration += Duration::from_nanos(timespec.tv_nsec as _);
-            Ok(Self::UNIX_EPOCH + duration)
-        } else {
-            Err(Error::new(
+        match chrono::Duration::new(timespec.tv_sec as _, timespec.tv_nsec as _) {
+            Some(duration) => Ok(Self::UNIX_EPOCH + duration),
+            None => Err(Error::new(
                 Ruby::get_with(val).exception_arg_error(),
-                "time must not be negative",
-            ))
+                "invalid timespec",
```

Comment:
This message will be raised to Ruby, but `timespec` isn't a thing at the Ruby level, so it won't make sense to a Ruby programmer.
The chrono docs say:

> Returns None when the duration is out of bounds

I think something like "time out of range" would fit with the kind of message Ruby raises in similar cases.